activerecord/test/cases/associations/has_one_through_disable_joins_associations_test.rb

```diff
@@ -16,6 +16,13 @@ def setup
     @member.reload
   end
 
+  # SponsorsTier - belongs_to sponsors_profile
+  # SponsorsProfile - belongs_to user
+  # SponsorsTier - has_one user through sponsors_profile disable_joins
```

Comment:
When I access `tier.user`, where `tier` is an instance of `SponsorsTier`, it's coming back `nil` unexpectedly. If I access `tier.sponsors_profile.user`, it is non-nil.